runtime/src/accounts.rs

```diff
@@ -547,14 +546,24 @@ impl Accounts {
                     })
                     .map(|(pubkey, account, _slot)| (*pubkey, account.lamports))
                 {
-                    collector.push(data)
+                    let index_of_first_smaller_account =
+                        collector.iter().position(|&(entry_pubkey, entry_balance)| {
+                            entry_balance < data.1
+                                || (entry_balance == data.1 && entry_pubkey > data.0)
```

Comment:
Looks like that pubkey comparison should add determinism. That's great, but is there a test for that case? Why `>`, when the balance comparison is `<`?  The difference would prevent you from replacing this implementation with a `BinaryHeap`.

Reply:
Ultimately, `>` vs `<` is arbitrary ... I'm not wedded to asc pubkeys. I'll rework with BinaryHeap!